src/cuga/backend/cuga_graph/nodes/cuga_lite/cuga_lite_graph.py

```
@@ -283,13 +324,161 @@ async def _indexed_tools_for_native_bind(
+    target_k = max_count - reserve
+    if target_k <= 0:
+        raise RuntimeError(
+            f"cuga_lite_bind_tools_max_count={max_count} is too small to fit even find_tools "
+            f"(reserve={reserve}). Raise the cap."
+        )
+
+    all_apps: List[Any] = []
+    if tool_provider is not None:
+        try:
+            all_apps = await tool_provider.get_apps()
+        except Exception as e:
```

Ruff (0.15.12)

[warning] Do not catch blind exception: `Exception`

(BLE001)